### ISSUE TITLE
Fix S3949 failures on test projects

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#2922]
### Description

There are some failing tests for the CBDE integration (rule S3649) - related to #619 

The failing projects are benchmark-dot-net, nodatime, ravendb and servuo

```
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: CBDE external process reported an error'. [C:\Project\samples\BenchmarkDotNet.Samples\BenchmarkDotNet.Samples.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: CBDE external process reported an error'. [C:\Project\samples\BenchmarkDotNet.Samples\BenchmarkDotNet.Samples.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: CBDE external process reported an error'. [C:\Project\samples\BenchmarkDotNet.Samples\BenchmarkDotNet.Samples.csproj]
```
```
CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Tools.DumpTimeZoneInfo\NodaTime.Tools.DumpTimeZoneInfo.csproj]
CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Tools.Common\NodaTime.Tools.Common.csproj]
CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.TzValidate.NzdCompatibility\NodaTime.TzValidate.NzdCompatibility.csproj]
CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime\NodaTime.csproj]
CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Tools.ValidateHistoricalNzd\NodaTime.Tools.ValidateHistoricalNzd.csproj]
CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.NzdPrinter\NodaTime.NzdPrinter.csproj]
CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Testing\NodaTime.Testing.csproj]
CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Benchmarks\NodaTime.Benchmarks.csproj]
CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.TzdbCompiler\NodaTime.TzdbCompiler.csproj]
CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Benchmarks\NodaTime.Benchmarks.csproj]
CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Benchmarks\NodaTime.Benchmarks.csproj]
CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.TzValidate.NodaDump\NodaTime.TzValidate.NodaDump.csproj]
CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Benchmarks\NodaTime.Benchmarks.csproj]
```

```
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: CBDE external process reported an error'. [C:\Project\src\Sparrow\Sparrow.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: CBDE external process reported an error'. [C:\Project\src\Sparrow\Sparrow.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: CBDE external process reported an error'. [C:\Project\src\Voron\Voron.csproj]
```

```
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: CBDE external process reported an error'. [C:\Project\Scripts\Scripts.csproj]

```

### Related information

* SonarC# Version - master branch (8.2 release candidate)

**Comment by andrei-epure-sonarsource** [comment]
reopening as `servuo` and `nodatime` analysis still fails, with similar exceptions

nodatime
```
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Tools.DumpTimeZoneInfo\NodaTime.Tools.DumpTimeZoneInfo.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Tools.Common\NodaTime.Tools.Common.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.TzValidate.NzdCompatibility\NodaTime.TzValidate.NzdCompatibility.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime\NodaTime.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Tools.ValidateHistoricalNzd\NodaTime.Tools.ValidateHistoricalNzd.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.NzdPrinter\NodaTime.NzdPrinter.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Testing\NodaTime.Testing.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.TzdbCompiler\NodaTime.TzdbCompiler.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Benchmarks\NodaTime.Benchmarks.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.TzValidate.NodaDump\NodaTime.TzValidate.NodaDump.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Benchmarks\NodaTime.Benchmarks.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Benchmarks\NodaTime.Benchmarks.csproj]
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: Object reference not set to an instance of an object.'. [C:\Project\src\NodaTime.Benchmarks\NodaTime.Benchmarks.csproj]
```

Servuo
```
    CSC : error AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CbdeHandlerRule' threw an exception of type 'SonarAnalyzer.CBDE.CbdeException' with message 'Top level error in CBDE handling: CBDE external process reported an error'. [C:\Project\Scripts\Scripts.csproj]
```

**Comment by andrei-epure-sonarsource** [comment]
it seems it's an intermittend failure
example logs from 31st of Dec https://api.cirrus-ci.com/v1/task/5017455420506112/logs/perftest.log

**Comment by andrei-epure-sonarsource** [comment]
this should be fixed by #2956 , we need to rerun the cirrus ci jobs

**Comment by andrei-epure-sonarsource** [comment]
the jobs ran successfully, I am closing this issue
